openedx/core/djangoapps/models/course_details.py

```diff
@@ -89,12 +89,13 @@ def fetch_about_attribute(cls, course_key, attribute):
         return value
 
     @classmethod
-    def fetch(cls, course_key):
+    def fetch(cls, course_key, descriptor=None):
```

Comment:
Create a new @classmethod named `populate` which takes the course descriptor as a parameter and encapsulates the part of this `fetch` function that sets the values on the CourseDetails object. Refactor `fetch` to do the course lookup `descriptor = modulestore().get_course(course_key)` and then call populate passing it the descriptor.